corehq/apps/sso/tests/test_forms.py

```diff
@@ -280,6 +280,52 @@ def test_is_active_has_met_requirements_and_value_updates(self, *args):
         idp = IdentityProvider.objects.get(id=self.idp.id)
         self.assertTrue(idp.is_active)
 
+    def test_cache_cleared_when_is_active_status_changes(self, *args):
```

Comment:
Similar concerns about this being at the wrong level. I think the concept we want to test is whether a save clears the cache, which could be tested at that low level (without as much setup). That would then eliminate the need to have the same tests for both `SSOEnterpriseSettingsForm` and `EditIdentityProviderAdminForm`.

Reply:
I think clearing the cache on every save is excessive when it's not always needed. The tests have to focus on these forms because that's where the cache clearing has to be triggered.

Reply:
Theoretically, instead of `SSOEnterpriseSettingsForm` and `EditIdentityProviderAdminForm`, let's say we had ten different forms all of which used this cache-clearing behavior. The cache clearing would still need to be triggered for all of these forms, but would we still write 5 duplicate tests just to verify that it happens for each of them? If the cache-clearing behavior is the same, I'd argue it makes more sense to write a single test verifying that saving the model causes the cache to clear.

Does that make sense? Is this just a case that it is reasonable to write the same test for 2 forms, but not for 5+? Or should we prefer duplicate tests regardless of the number of forms?

Reply:
reverted, so no longer applicable

Reply:
I agree though on the point about having to create more essentially duplicated tests being not great and one thing that did annoy me about the original approach...but I was writing for optimization, though arguably it wasn't necessary at that point.